### ISSUE TITLE
conda installation of 3.2.2 in uvcdat 2.8.0 requests a downgrade of libnetcdf

### ISSUE THREAD

**Issue by jypeter** [PCMDI/cmor#138]
Congratulations @dnadeau4 on getting this new update! But if I try to install it in uvcdat 2.8.0, it wants to **downgrade libnetcdf**...

What should I do? Force the installation and keep the libnetcdf that's already there? Or wait? I don't need CMOR right now. It's just that I have started practicing with cmor and I wanted to check what happens if I try to install it

```
$ conda install -n uvcdat-2.8.0 -c conda-forge -c pcmdi -c uvcdat cmor
Fetching package metadata ...............
Solving package specifications: .

Package plan for installation in environment /home/share/unix_files/cdat/miniconda2/envs/uvcdat-2.8.0:

The following NEW packages will be INSTALLED:

    cmor:      3.2.2-np111py27_0 pcmdi

The following packages will be DOWNGRADED due to dependency conflicts:

    libnetcdf: 4.4.1.1-2         conda-forge --> 4.4.1-0 conda-forge

```

**Comment by dnadeau4** [comment]
I used the same conda-recipes as uvcdat-2.8.0.   I looked at the meta.yaml and they don't seem to `pin` the version and conda-forge select libnecdf-4.4.1-0.  https://github.com/conda-forge/staged-recipes/wiki/Pinned-dependencies

I don't know how `uv-cdat` created their package with `libnetcdf 4.4.1.1-2`.

**Comment by dnadeau4** [comment]
@jypeter You can try to create a new environment until I figure that one out.   I looked at all our `meta.yaml` file and it is not obvious why `uv-cdat` is using `libnetcdf 4.4.1.1-2`.   It might just be related to their version of `hdf5`.

```
conda create -n cmor-3.2.2 -c conda-forge -c pcmdi -c uvcdat cmor
source activate cmor-3.2.2
```
@doutriaux1 Your help would be welcome here.

**Comment by dnadeau4** [comment]
@jypeter You can also try to create a new UV-CDAT environment to see if conda is getting libnetcdf 4.4.1.1-2.  (You can answer `no` to the question.  This is a way  to see what `conda` from the archive pull without creating the environment.)

**Comment by jypeter** [comment]
@dnadeau4 I get the following (with  libnetcdf **4.4.1-0**) if I try a _dry-run_ installation oc cmor
```
bash-4.1$ conda create --dry-run -n cmor-3.2.2 -c conda-forge -c pcmdi -c uvcdat cmor
Fetching package metadata ...............
Solving package specifications: .

Package plan for installation in environment /home/share/unix_files/cdat/miniconda2/envs/cmor-3.2.2:

The following NEW packages will be INSTALLED:

    ca-certificates: 2017.1.23-0                       conda-forge
    cdat_info:       2.8-py27_0                        conda-forge
    cdms2:           2.8.1-np111py27_0                 uvcdat
    cdtime:          2.8-np111py27_0                   conda-forge
    certifi:         2017.1.23-py27_0                  conda-forge
    clapack:         3.2.1-0                           conda-forge
    cmor:            3.2.2-np111py27_0                 pcmdi
    curl:            7.52.1-0                          conda-forge
    distarray:       2.8-py27_0                        conda-forge
    esmf:            ESMF_6_3_0rp1_ESMP_01-np111py27_2 uvcdat
    expat:           2.1.0-2                           conda-forge
    g2clib:          1.6.0-2                           conda-forge
    hdf4:            4.2.12-0                          conda-forge
    hdf5:            1.8.17-9                          conda-forge
    jasper:          1.900.1-4                         conda-forge
    jpeg:            9b-0                              conda-forge
    lapack:          3.6.1-1                           conda-forge
    libcdms:         2.8-0                             conda-forge
    libcf:           1.0-py27_0                        conda-forge
    libdrs:          2.8-0                             conda-forge
    libdrs_f:        2.8-py27_0                        conda-forge
    libgcc:          5.2.0-0
    libgfortran:     3.0.0-1
    libnetcdf:       4.4.1-0                           conda-forge
    libpng:          1.6.28-0                          conda-forge
    libtiff:         4.0.6-7                           conda-forge
    mkl:             2017.0.1-0
    ncurses:         5.9-10                            conda-forge
    numpy:           1.11.3-py27_0
    openssl:         1.0.2h-3                          conda-forge
    ossuuid:         1.6.2-0                           conda-forge
    pip:             9.0.1-py27_0                      conda-forge
    python:          2.7.12-2                          conda-forge
    readline:        6.2-0                             conda-forge
    setuptools:      33.1.0-py27_0                     conda-forge
    sqlite:          3.13.0-1                          conda-forge
    tk:              8.5.19-1                          conda-forge
    udunits2:        2.2.23-0                          conda-forge
    wheel:           0.29.0-py27_0                     conda-forge
    xz:              5.2.2-0                           conda-forge
    zlib:            1.2.11-0                          conda-forge
```
You can check the following page if you want to see my (quite) full notes about 2.8.0 installation: https://wiki.lsce.ipsl.fr/pmip3/doku.php/other:uvcdat:cdat_conda:cdat_2_8_0

Or at least the following link that lists what was installed for 2.8.0 (with libnetcdf **4.4.1.1-2**): https://wiki.lsce.ipsl.fr/pmip3/lib/exe/fetch.php/other:uvcdat:uvcdat-2.8.0_install_dry_with-forge_160220.txt

Like I said, I don't need cmor now, so there is **no rush at all**. Just wanted to report the possibly odd behavior

**Comment by dnadeau4** [comment]
We will create 2 packages PrePARE and CMOR. This might not solve this issue.

**Comment by jypeter** [comment]
OK, let me know if you want me to test something

**Comment by dnadeau4** [comment]
@doutriaux1 @taylor13 and I have been talking about it today,  if I push cmor into conda-forge, that should fix the issue.  CDMS2 is already in the conda-forge channel with all its dependencies.

**Comment by cehbrecht** [comment]
@dnadeau4 
i didn't find the cmor conda recipe but from the downloaded conda package it looks like you have pinned the libnetcdf version: ``libnetcdf 4.4.1``.

The highest version on the conda default channel seems to be 4.4.1 while on conda-forge it is 4.4.1.1. We use several packages from conda-forge, like cdo, which are build with the latest version from the conda-forge channel. When you look at the cdo recipe on conda-forge, they use the following libnetcdf pin: 
```
libnetcdf 4.4.*
```
https://github.com/conda-forge/cdo-feedstock/blob/master/recipe/meta.yaml#L28

**Comment by dnadeau4** [comment]
@cehbrecht I tried with `libnetcdf 4.4.*` and it worked I got `libnetcdf 4.4.1.1-4`.  :smile: 
Thanks!

**Comment by durack1** [comment]
@dnadeau4 I'm getting the same error:
```
(cdatcmorNnclnco) durack1@oceanonly:[150219_AMIPForcingData]:[4772]> conda install -c conda-forge -c pcmdi/label/nightly cmor
Fetching package metadata .............
Solving package specifications: .
Package plan for installation in environment /export/durack1/anaconda2/envs/cdatcmorNnclnco:
The following NEW packages will be INSTALLED:
    cmor:      2017.04.18.3.2.3-np111py27_0 pcmdi/label/nightly
The following packages will be DOWNGRADED due to dependency conflicts:
    libnetcdf: 4.4.1.1-4                    conda-forge         --> 4.4.1-0 conda-forge
```
The environment already has a `uvcdat` install using `conda create -n cdatcmorNnclnco -c conda-forge -c uvcdat uvcdat`. It'd be good not to force the downgrade, the netcdf update fixed a couple of things

**Comment by dnadeau4** [comment]
fixed in cmor 3.2.3

**Comment by jypeter** [comment]
Thanks! I have now installed it successfully in 2.8.0 (but not used/tested it)

**Comment by durack1** [comment]
@dnadeau4 I can also confirm that it is fixed in `3.2.3`
